internal/services/extendedlocation/extended_location_custom_location_test.go

```diff
@@ -0,0 +1,351 @@
+  name              = "extension-%[1]d"
+  cluster_id        = azurerm_arc_kubernetes_cluster.test.id
+  extension_type    = "microsoft.vmware"
+  release_namespace = "vmware-extension"
+
+  configuration_settings = {
+    "Microsoft.CustomLocation.ServiceAccount" = "vmware-operator"
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  depends_on = [
+    azurerm_linux_virtual_machine.test
+  ]
+}
+`, data.RandomInteger, data.Locations.Primary, credential, publicKey, provisionTemplate)
+}
+
+func (r CustomLocationResource) generateKey() (string, string, error) {
+	privateKey, err := rsa.GenerateKey(cryptoRand.Reader, 4096)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to generate RSA key")
+	}
+
+	privateKeyBytes := x509.MarshalPKCS1PrivateKey(privateKey)
+	privateKeyBlock := &pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: privateKeyBytes,
+	}
+
+	privatePem := pem.EncodeToMemory(privateKeyBlock)
+	if privatePem == nil {
+		return "", "", fmt.Errorf("failed to encode pem")
+	}
+
+	return string(privatePem), base64.StdEncoding.EncodeToString(x509.MarshalPKCS1PublicKey(&privateKey.PublicKey)), nil
+}
+
+func (r CustomLocationResource) getCredentials(t *testing.T) (credential, privateKey, publicKey string) {
+	// generateKey() is a time-consuming operation, we only run this test if an env var is set.
+	if os.Getenv(resource.EnvTfAcc) == "" {
+		t.Skipf("Acceptance tests skipped unless env '%s' set", resource.EnvTfAcc)
+	}
+
+	credential = fmt.Sprintf("P@$$w0rd%d!", rand.Intn(10000))
+	privateKey, publicKey, err := r.generateKey()
+	if err != nil {
+		t.Fatalf("failed to generate key: %+v", err)
+	}
+
+	return
+}
+
+func (r CustomLocationResource) provisionTemplate(data acceptance.TestData, credential string, privateKey string) string {
+	return fmt.Sprintf(`
+connection {
+ type     = "ssh"
+ host     = azurerm_public_ip.test.ip_address
+ user     = "adminuser"
+ password = "%[1]s"
+}
+
+provisioner "file" {
+ content = templatefile("testdata/install_agent.sh.tftpl", {
+   subscription_id     = "%[4]s"
+   resource_group_name = azurerm_resource_group.test.name
+   cluster_name        = "acctest-akcc-%[2]d"
+   location            = azurerm_resource_group.test.location
+   tenant_id           = "%[5]s"
+   working_dir         = "%[3]s"
+ })
+ destination = "%[3]s/install_agent.sh"
+}
+
+provisioner "file" {
+ source      = "testdata/install_agent.py"
+ destination = "%[3]s/install_agent.py"
+}
+
+provisioner "file" {
+ source      = "testdata/kind.yaml"
+ destination = "%[3]s/kind.yaml"
+}
+
+provisioner "file" {
+ content     = <<EOT
+%[6]s
+EOT
+ destination = "%[3]s/private.pem"
+}
+
+provisioner "remote-exec" {
+ inline = [
+   "sudo sed -i 's/\r$//' %[3]s/install_agent.sh",
+   "sudo chmod +x %[3]s/install_agent.sh",
+   "bash %[3]s/install_agent.sh > %[3]s/agent_log",
+ ]
+}
+`, credential, data.RandomInteger, "/home/adminuser", os.Getenv("ARM_SUBSCRIPTION_ID"), os.Getenv("ARM_TENANT_ID"), privateKey)
```

Comment:
Please try to avoid consuming environment variables unless there is no other way to obtain information. In this case, we should use an `azurerm_client_config` data source to obtain the subscription ID and tenant ID for the current provider instance.